### PR TITLE
catt: relax click dependency

### LINKED ISSUE #132784: python3Packages.catt broken due to invalid dependencies
### Describe the bug
catt fails to build with - `ERROR: No matching distribution found for Click<8,>=7.1.2`

### Steps To Reproduce

1. `nix build nixpkgs#catt`

### Expected behavior

catt builds correctly

### Notify maintainers

@dtzWill

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Closes #132784

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/catt/default.nix [hunk 1]
```diff
@@ -19,6 +19,11 @@ buildPythonApplication rec {
     youtube-dl
   ];
 
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace 'Click>=7.1.2,<8' 'click' \
```
Comment (fabaff) [inline pkgs/applications/video/catt/default.nix:24]: `click>8` contains breaking changes. It would probably be better to override `click`.

Comment (Artturin) [reply pkgs/applications/video/catt/default.nix:24]: https://github.com/skorokithakis/catt/pull/311
it was pinned because it didn't work with python 3.5

Comment (Artturin) [reply pkgs/applications/video/catt/default.nix:24]: opened an issue  #354

**Comment by dtzWill** [comment]
Thanks!

**Pull request merged**